src/Container.php

```diff
@@ -209,64 +201,46 @@ private function checkForUnionType(
                 $paramType->getTypes(),
             );
 
-            $this->throwNonResolvableParameterTypeException(
-                $qn,
-                $param,
-                reason: 'It has union type:' . implode('|', $typeNames),
+            $message = sprintf(
```

Comment:
Error messages we use for other exceptions don't make sense for union types because the parameter type being mentioned and making the exception message too long. The first version didn't have this issue. I didn't realize this while creating `throwNonResolvableParameterTypeException`

E.g.: `SomeVendor\SomePackage\ClassName|SomeVendor\SomePackage\AnotherClassName`

Just reverted back to its original state basically